File: hackerearth-vignan/III-SEM/challenge_07/pattren.cpp
#include <iostream>
#include <string>

int main() {
    int numTests;
    std::cin >> numTests;

    for (int i = 0; i < numTests; i++) {
        int n;
        std::cin >> n;
        int space = (n-4)/2;
        // Print the top row
        std::cout << std::string(n, '*') << std::endl;

        // Print the middle section
        for (int j = 0; j < space; j++) {
            
            std::cout << '*' << std::string(space, ' ') << '*' << ' ' << '*' << std::string(space, ' ') << '*' << std::endl;
        }

        // Print the bottom row
        std::cout << std::string(n, '*') << std::endl;

        // Print the middle part
        std::cout << '*' << std::string(space, ' ') << '*' << ' ' << '*' << std::string(space, ' ') << '*' << std::endl;

        // Print the bottom part
        std::cout << std::string(n, '*') << std::endl;

        // Print the middle section
        for (int j = 0; j < space; j++) {

            std::cout << '*' << std::string(space, ' ') << '*' << ' ' << '*' << std::string(space, ' ') << '*' << std::endl;
        }

        // Print the bottom row
        std::cout << std::string(n, '*') << std::endl;

        // Print a blank line
        std::cout << std::endl;
    }

    return 0;
}

File: hackerearth-vignan/III-SEM-2022-26/challenge_07/pattren.py

for _ in range(int(input())):
    # Print the top row
    n = int(input())
    print('*' * n)

    # Print the middle section
    for _ in range((n-1)//2):
        space = (n//2) - 2
        print('*' + ' ' * space + '*' + ' ' + '*' + ' ' * space + '*')

    # Print the bottom row
    print('*' * n)

    # print the middle part
    print('*' + ' ' * space + '*' + ' ' + '*' + ' ' * space + '*')

    # bottom part
    print('*' * n)

    # Print the middle section
    for _ in range((n-1)//2):
        space = (n//2) - 2
        print('*' + ' ' * space + '*' + ' ' + '*' + ' ' * space + '*')

    # Print the bottom row
    print('*' * n)

    # Print a blank line
    print()
